### ISSUE TITLE
`CODEOWNERS` and branches - organization & delegation

### ISSUE THREAD

**Issue by xeho91** [terminal-nerds/terminal-nerd#22]
# CODEOWNERS

**This issue starts the discussion on how to organize the responsibilities delegation for this project.**

---

## Initial proposal

A code owner is a person responsible for the code review and required to accept the changes.

### Directories

- `client/`, aka **front-end** - @mariushilleke as a core code owner
- `server/` aka **back-end** - @lax0248917 as a core code owner

### Branches

Conditions for managing the operations in the following branches
- `main` - requires **everyone's** approval (@xeho91, @mariushilleke, @lax0248917)
- `develop` requires **one** approval from the person responsible for the directory where the changes have been made
  OR, in case of absence, to prevent blocking the process - @xeho91 approval.

---

Thoughts?

Please comment that you agree, approve the changes before we set it up.

**Comment by xeho91** [comment]
**Question: How do we configure [`helpers/`](./helpers) directory?**

Since they're supposed to help both sides, in my opinion, they need approval from the two of you (@lax0248917 and @mariushilleke).

**Comment by lax0248917** [comment]
Above you have a breakdown for the client and the server, can't you annotate codeowner requirements for the helper and assign a requirement that both of us approve? I have to look at how the code owner config file is setup to familiarize myself with it. Its been awhile since we used it last.

**Comment by lax0248917** [comment]
@xeho91 now that I think about it, we had this same question come up with Team Armadillo and our last project. As it doesn't look like you can use the CODEOWNERS config file to mandate two approvals, we created a Team for the organization and then required the team to approve. I think this was a successful workaround but may be overkill for this project.

**Comment by xeho91** [comment]
> @xeho91 now that I think about it, we had this same question come up with Team Armadillo and our last project. As it doesn't look like you can use the CODEOWNERS config file to mandate two approvals, we created a Team for the organization and then required the team to approve. I think this was a successful workaround but may be overkill for this project.

Indeed, creating teams inside our GitHub organization would be overkill for now until we find extra people.

**Comment by lax0248917** [comment]
So then maybe there is a way to use a github actions to add necessary reviewers based on the whether code changes are being made in the /helper directory
